### PR TITLE
[Feature] Set up unit test that loads activity from exported json file #3807

### PR DESCRIPTION
**Pull request by dhaneshwaritendle**
image

-Created file portfolio-calculator-dynamic-buy-and-sell.spec.ts  which is in directory apps/api/src/app/portfolio/calculator/twr/portfolio-calculator-dynamic-buy-and-sell.spec.ts

-It checks for the 'activities' in which json data is parsed 

-It passed the test suites 

If I have missed something I would be happy to rectify it.

### CODE REVIEW

File: apps/api/src/app/portfolio/calculator/twr/portfolio-calculator-dynamic-buy-and-sell.spec.ts [hunk 1]
```diff
@@ -0,0 +1,250 @@
+import { Activity } from '@ghostfolio/api/app/order/interfaces/activities.interface';
+import {
+  activityDummyData,
```
Comment (dtslvr) [inline apps/api/src/app/portfolio/calculator/twr/portfolio-calculator-dynamic-buy-and-sell.spec.ts:3]: Please remove unused import

File: apps/api/src/app/portfolio/calculator/twr/portfolio-calculator-dynamic-buy-and-sell.spec.ts [hunk 2]
```diff
@@ -0,0 +1,250 @@
+import { Activity } from '@ghostfolio/api/app/order/interfaces/activities.interface';
+import {
+  activityDummyData,
+  symbolProfileDummyData,
```
Comment (dtslvr) [inline apps/api/src/app/portfolio/calculator/twr/portfolio-calculator-dynamic-buy-and-sell.spec.ts:4]: Please remove unused import

File: apps/api/src/app/portfolio/calculator/twr/portfolio-calculator-dynamic-buy-and-sell.spec.ts [hunk 3]
```diff
@@ -0,0 +1,250 @@
...
+import { ExchangeRateDataService } from '@ghostfolio/api/services/exchange-rate-data/exchange-rate-data.service';
+import { PortfolioSnapshotService } from '@ghostfolio/api/services/queues/portfolio-snapshot/portfolio-snapshot.service';
+import { PortfolioSnapshotServiceMock } from '@ghostfolio/api/services/queues/portfolio-snapshot/portfolio-snapshot.service.mock';
+import { parseDate } from '@ghostfolio/common/helper';
+
+import { Big } from 'big.js';
+import * as fs from 'fs';
+import { last } from 'lodash';
+
+import path = require('path');
```
Comment (dtslvr) [inline apps/api/src/app/portfolio/calculator/twr/portfolio-calculator-dynamic-buy-and-sell.spec.ts:25]: Please import like this:
```
import { join } from 'path';
```

File: apps/api/src/app/portfolio/calculator/twr/portfolio-calculator-dynamic-buy-and-sell.spec.ts [hunk 4]
```diff
@@ -0,0 +1,250 @@
...
+import { RedisCacheService } from '@ghostfolio/api/app/redis-cache/redis-cache.service';
+import { RedisCacheServiceMock } from '@ghostfolio/api/app/redis-cache/redis-cache.service.mock';
+import { ConfigurationService } from '@ghostfolio/api/services/configuration/configuration.service';
+import { ExchangeRateDataService } from '@ghostfolio/api/services/exchange-rate-data/exchange-rate-data.service';
+import { PortfolioSnapshotService } from '@ghostfolio/api/services/queues/portfolio-snapshot/portfolio-snapshot.service';
+import { PortfolioSnapshotServiceMock } from '@ghostfolio/api/services/queues/portfolio-snapshot/portfolio-snapshot.service.mock';
+import { parseDate } from '@ghostfolio/common/helper';
+
+import { Big } from 'big.js';
+import * as fs from 'fs';
```
Comment (dtslvr) [inline apps/api/src/app/portfolio/calculator/twr/portfolio-calculator-dynamic-buy-and-sell.spec.ts:22]: Please import like this:
```
import { existsSync, readFileSync } from 'fs';
```

File: apps/api/src/app/portfolio/calculator/twr/portfolio-calculator-dynamic-buy-and-sell.spec.ts [hunk 5]
```diff
@@ -0,0 +1,250 @@
...
+      configurationService,
+      currentRateService,
+      exchangeRateDataService,
+      portfolioSnapshotService,
+      redisCacheService
+    );
+  });
+
+  //read from activities json
+  let parsedData;
```
Comment (dtslvr) [inline apps/api/src/app/portfolio/calculator/twr/portfolio-calculator-dynamic-buy-and-sell.spec.ts:91]: `let activities;`

Comment (dtslvr) [reply apps/api/src/app/portfolio/calculator/twr/portfolio-calculator-dynamic-buy-and-sell.spec.ts:91]: Add line break

File: apps/api/src/app/portfolio/calculator/twr/portfolio-calculator-dynamic-buy-and-sell.spec.ts [hunk 6]
```diff
@@ -0,0 +1,250 @@
...
+  beforeAll(() => {
+    const jsonFilePath = path.join(__dirname, '../../../../../../../test/import/ok-novn-buy-and-sell.json');
+    if (!fs.existsSync(jsonFilePath)) throw new Error('JSON file not found at: ' + jsonFilePath);
+    const jsonData = fs.readFileSync(jsonFilePath, 'utf8');
+    parsedData = JSON.parse(jsonData);
```
Comment (dtslvr) [inline apps/api/src/app/portfolio/calculator/twr/portfolio-calculator-dynamic-buy-and-sell.spec.ts:96]: `activities = JSON.parse(jsonData).activities;`

File: apps/api/src/app/portfolio/calculator/twr/portfolio-calculator-dynamic-buy-and-sell.spec.ts [hunk 7]
```diff
@@ -0,0 +1,250 @@
...
+  });
+  describe('get current positions', () => {
```
Comment (dtslvr) [inline apps/api/src/app/portfolio/calculator/twr/portfolio-calculator-dynamic-buy-and-sell.spec.ts:98]: Add line break

**Review by dtslvr — CHANGES_REQUESTED** [review]
Thank you for the contribution, @dhaneshwaritendle! I have added a few comments.

File: apps/api/src/app/portfolio/calculator/twr/portfolio-calculator-dynamic-buy-and-sell.spec.ts (hunk 7 above)
Comment (dhaneshwaritendle) [reply apps/api/src/app/portfolio/calculator/twr/portfolio-calculator-dynamic-buy-and-sell.spec.ts:98]: on it!

**Review by dtslvr — APPROVED** [review]
Great work, @dhaneshwaritendle! 👌🏻
I have extracted the functionality to load the json file to a helper.
Ready to merge now.

**Pull request merged**